torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -950,6 +968,23 @@ def call_function(
             should_flatten_outputs=True,
         )
 
+        subgraph_example_value = [
+            proxy.node.meta["example_value"] for proxy in body_r.as_proxy()
+        ]
+
+        with tx.output.fake_mode:
+            # We need to expand the example output from map() so that it has
+            # the same first dimension as the mapped input.
+            # We also do a clone with contiguous_format. This is to be consistent with
+            # eager semantic of map, which stacks the outputs. The result is contiguous
+            # as a result of the stack operation.
```

Comment:
It's actually kind of surprising you don't try to preserve the inner layout (e.g., to preserve NHWC'ness, if necessary). I am not doubting that this is right, but maybe the eager semantics are not optimal either lol

Reply:
yeah, we can probably address it in the future when we see a real use case.